### PR TITLE
fix: [CI-4581]: Skip test selection and cg collection in case of manual run

### PR DESCRIPTION
**Pull request by rutvijmehta-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/37235)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/37235)

File: product/ci/addon/tasks/runtests.go [hunk 1]
```diff
@@ -282,6 +281,37 @@ func valid(tests []types.RunnableTest) bool {
...
+		// PR execution: Call TI svc only when there is a chance of running selected tests
+		resp, err := selectTestsFn(ctx, files, r.runOnlySelectedTests, r.id, r.log, r.fs)
+		if err != nil {
+			log.Errorw("there was some issue in trying to intelligently figure out tests to run. Running all the tests", "error", zap.Error(err))
+			r.runOnlySelectedTests = false
+		} else if !valid(resp.Tests) { // This shouldn't happen
+			log.Errorw("test intelligence did not return suitable tests")
+			r.runOnlySelectedTests = false
+		} else if resp.SelectAll == true {
+			log.Infow("intelligently determined to run all the tests")
```
Comment (hemanthmantri) [inline product/ci/addon/tasks/runtests.go:306]: Let's remove `intelligently`

File: product/ci/addon/tasks/runtests.go [hunk 2]
```diff
@@ -282,6 +281,37 @@ func valid(tests []types.RunnableTest) bool {
...
+			r.runOnlySelectedTests = false
+		} else {
+			r.log.Infow(fmt.Sprintf("intelligently running tests: %s", resp.Tests))
```
Comment (hemanthmantri) [inline product/ci/addon/tasks/runtests.go:309]: `selected tests: %s`

File: product/ci/addon/tasks/runtests.go [hunk 3]
```diff
@@ -282,6 +281,37 @@ func valid(tests []types.RunnableTest) bool {
 	return true
 }
 
+func (r *runTestsTask) getTestSelection(ctx context.Context, files []types.File, isManual bool) types.SelectTestsResp {
+	resp := types.SelectTestsResp{}
+	log := r.log
+
+	if isManual {
+		// Manual execution: Select all tests in case of manual execution
+		log.Infow("detected manual execution - for intelligence to be configured, a PR must be raised. Running all the tests")
```
Comment (hemanthmantri) [inline product/ci/addon/tasks/runtests.go:290]: s/intelligence/test intelligence

**Review by hemanthmantri — APPROVED** [review]

**Review by jamie-harness — APPROVED** [review]
> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37235)* status: :shipit: complete! ✓

File: product/ci/addon/tasks/runtests.go (hunk 3 above)
Comment (rutvijmehta-harness) [reply product/ci/addon/tasks/runtests.go]: Done

File: product/ci/addon/tasks/runtests.go (hunk 1 above)
Comment (rutvijmehta-harness) [reply product/ci/addon/tasks/runtests.go]: Done

File: product/ci/addon/tasks/runtests.go (hunk 2 above)
Comment (rutvijmehta-harness) [reply product/ci/addon/tasks/runtests.go]: Done

**Comment by rutvijmehta-harness** [comment]
trigger messagecheck

**Review by jamie-harness — APPROVED** [review]
> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37235)* status: 3. unresolved discussions.

**Review by hemanthmantri — APPROVED** [review]

**Review by hemanthmantri — APPROVED** [review]
> Reviewed 1 of 3 files at r1, 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37235)* status: :shipit: complete! ✓

**Pull request merged**